test/console/graphql/queries/observability_queries_test.exs

```diff
@@ -192,6 +196,54 @@ defmodule Console.GraphQl.ObservabilityQueriesTest do
       assert line["log"] == "a log"
     end
 
+    test "it can fetch logs from an elasticsearch index" do
+      user = insert(:user)
+      svc = insert(:service, read_bindings: [%{user_id: user.id}])
+
+      # We want a fresh index so delete the index if it already exists
+      # Just in case test fails we want to make sure next run doesn't reuse old index
+      with true <- ES.index_exists?(@elastic_cluster_url, @elastic_index_name) do
+        ES.delete_index(@elastic_cluster_url, @elastic_index_name)
+      end
+
+      ES.create_index(@elastic_cluster_url, @elastic_index_name)
+
+      ES.index_documents(@elastic_cluster_url, @elastic_index_name, [
+        %{
```

Comment:
let's move these documents to helper function in elastic utils, eg something like:

```elixir
def log_document(%Service{}, line) do
  ...map data for log
end
```

And that way you can just index by doing something like:

```elixir
log_document(svc, "valid log") |> index()
log_document(svc, "valid log2") |> index()
log_document(svc2, "valid log") |> index()
refresh()
```

Reply:
you should also just import the elastic helper so it's quicker to call the functions I think